portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -103,4 +114,19 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     // Redirect back to the contact page.
     response.sendRedirect("/contact_me.html");
   }
+
+  /** Returns the username of the user with id, or null if the user has not set a username. */
+  private String getUsername(String id) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query =
+        new Query("UserInfo")
+            .setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+    PreparedQuery results = datastore.prepare(query);
```

Comment:
This naming seems a bit off. The object type is "PreparedQuery" but you've named the variable "results". I would rename it.

Also, results is a bit vague here.